web/src/stream_edit_subscribers.ts

```diff
@@ -164,10 +207,10 @@ function subscribe_new_users({pill_user_ids}) {
         // case occurs when creating user pills from a stream.
         user_id_set.delete(current_user.user_id);
     }
-    let ignored_deactivated_users;
+    let ignored_deactivated_users: people.User[] = [];
     if (deactivated_users.size > 0) {
-        ignored_deactivated_users = [...deactivated_users];
-        ignored_deactivated_users = ignored_deactivated_users.map((user_id) =>
```

Comment:
`ignored_deactivated_users` : an array with elements of type  `number` was being assigned the `map`ped array containing element of type `people.User`. I created a new variable to separate both tasks.

Reply:
1. We are not using `people.User` import style for types right? I think we are importing `User` separately in other modules?
2. According to new code `ignored_deactivated_users` variable will always be an `array` as opposed to previous version where it could be `undefined`, have you checked that this does not causes any issue in the following code paths?

Reply:
1. Fixed. The module now uses `User` style only.
2. Yeah, `ignored_deactivated_users` can be undefined and template is designed accordingly.
```
{{#if ignored_deactivated_users}}
    {{#each ignored_deactivated_users}}
        {{#if @first}}
        {{t "Ignored deactivated users:" }}
        {{/if}}
        <a data-user-id="{{user_id}}" class="view_user_profile">{{full_name}}</a>{{#unless @last}},{{else}}.{{/unless}}
    {{/each}}
{{/if}}

```
 I changed its type declaration ie. `optional` as well as `undefined`, although the other `User` type parameters can actually never be undefined (will be optional or empty array at worst).